reserve_harvester.py

```diff
@@ -116,54 +178,18 @@ def process_wp_reserves():
     wp articles categorized as naturreservat that are missing
     in the source file (even with similar looking name).
     """
-    results_file_exact = "svwp_to_nature_id_exact.json"
-    results_file_none = "svwp_to_nature_id_none.json"
-    results_file_multiple = "svwp_to_nature_id_multiple.json"
-    results = []
-    results_none = []
-    results_multiple = []
-    reserves_on_wp = read_wp_nr_list()
-    reserves_source = read_reserve_csv()
-    article_count = len(reserves_on_wp)
-    print("Processing {} svwp articles.".format(article_count))
-    counter = 0
-    for article_title in reserves_on_wp:
-        counter = counter + 1
-        if counter % 10 == 0:
-            print("Processed {}/{}...".format(counter, article_count))
-        article_title = article_title.replace("_", " ")
-        if (not article_title.startswith("Lista") and
-                "nationalpark" not in article_title.lower()):
-            municipalities = get_municipalities(article_title)
-            if municipalities:
-                guesses = [x for x in
-                           reserves_source if
-                           (x["name"] == article_title or x["name"].startswith(
-                               article_title) or
-                            article_title.startswith(x["name"])) and
-                           x["municipalities"] == municipalities]
-                if len(guesses) == 1:
-                    entry = {}
-                    entry["wp_article"] = article_title
-                    entry["source_name"] = guesses[0]["name"]
-                    entry["nature_id"] = guesses[0]["nature_id"]
-                    entry["item"] = utils.q_from_wikipedia("sv", article_title)
-                    results.append(entry)
-                    utils.json_to_file(results_file_exact, results)
-                elif len(guesses) > 1:
-                    entry = {}
-                    entry["wp_article"] = article_title
-                    nature_ids = []
-                    for row in guesses:
-                        nature_ids.append(row["nature_id"])
-                    entry["nature_id"] = nature_ids
-                    results_multiple.append(entry)
-                    utils.json_to_file(results_file_multiple, results_multiple)
-                else:
-                    entry = {}
-                    entry["wp_article"] = article_title
-                    results_none.append(entry)
-                    utils.json_to_file(results_file_none, results_none)
+    guesses = []
+    article_title = article_title.replace("_", " ")
+    municipalities = get_municipalities(article_title)
+    if municipalities:
+        guesses = [x for x in
```

Comment:
I think breaking this into multiple parts (maybe also a loop instead of list comprehension) would make it more legible. But if so do it in a follow up patch.

`guesses = [x for x in reserves_source if guess_nature_reserve(x, municipalities, article_title)]`

with something like
```python
def guess_nature_reserve(x, municipalities, article_title):
    if (not x["municipalities"] == municipalities) or (not x["status"] == "Gällande"):
         return False
    elif: x["name"] == article_title or x["name"].startswith(article_title) or article_title.startswith(x["name"])):
        return True
    else:
         return False
```